### PR TITLE
enhancement(vector source): comply with component spec

### PR DESCRIPTION
**Pull request by jdrouet**
Following the issues #11036 and #9643

Signed-off-by: Jérémie Drouet <jeremie.drouet@gmail.com>

### CODE REVIEW

File: src/sinks/vector/v2/sink.rs [hunk 1]
```diff
@@ -28,10 +30,17 @@ pub struct VectorSink {
 impl VectorSink {
     async fn run_inner(self: Box<Self>, input: BoxStream<'_, Event>) -> Result<(), ()> {
         input
-            .map(|mut event| EventData {
-                byte_size: event.size_of(),
-                finalizers: event.take_finalizers(),
-                wrapper: EventWrapper::from(event),
+            .map(|mut event| {
+                let byte_size = event.size_of();
+                emit!(&EventsReceived {
```
Comment (jdrouet) [inline src/sinks/vector/v2/sink.rs:35]: I would have preferred doing this `emit` with batches of events to reduce the number of calls but the RFC says `immediately after creating or receiving one or more Vector events`.

Comment (jdrouet) [reply src/sinks/vector/v2/sink.rs:35]: I'd prefer calling it from here https://github.com/vectordotdev/vector/blob/master/src/sinks/vector/v2/service.rs#L102

**Review by jszwedko — COMMENTED** [review]
Thanks! I left a comment below but I think the event you added here is already being published which would result in double counting.

There does appear to one missing bit for the v2 `vector` sink though, `vector_component_sent_event_bytes_total`, as noted below.

The issues you linked to are actually about the `vector` _source_ rather than sink 😄 That is the one that seems to be missing most event processing metrics.

File: src/sinks/vector/v2/sink.rs (hunk 1 above)
Comment (jdrouet) [reply src/sinks/vector/v2/sink.rs:35]: Found the reason, dropping this commit as it's no longer needed 😉

**Comment by jdrouet** [comment]
@jszwedko with those changes, the events appear to be well published.

**Review by jszwedko — APPROVED** [review]
Awesome! Looks great! Thanks for tracking this one down.

**Pull request merged**